app/models/types/pattern.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+#-- copyright
+#++
+
+module Types
+  Pattern = Data.define(:pattern) do
```

Comment:
Naming questions because I'm that guy:

1. Pattern for the type name sounds okay to me, but I ask myself if it's specific enough. I don't want to call it `SubjectPattern`, but maybe there's a universe where we can call `NamePattern` or `AttributePattern`.
2. We might going to have situations where we say `pattern.pattern`. Should we name the attribute `value` or something generic?

Reply:
Fair enough. So, about the name of the type first. The something in `SomethingPattern` could be:
- **Attribute**: because we are assigning attributes to a pattern. But not sure if it's meaningful enough
- **WorkPackageNamePattern**: more specific, but probably too enterprisified to be meaningful.

Could not think of anything else now. I'll be happy if we stay with Pattern as well, just trying to bounce random ideas.

Reply:
About the attribute ` :pattern` inside the type `Pattern`. Absolutely fine. I don't like calling `thing.thing`, but at the same time it's not a big deal.

Reply:
@brunopagno one good thing is that those objects are created by the PatternType, so we can make changes to their internals at will, even after they get created.

Reply:
That works well for me <3